### PR TITLE
Fix DTIPrep pipeline when create NIfTI is set to yes

### PR DESCRIPTION
**Pull request by cmadjar**
### Description

When a project uses NIfTI files creation, the DTIPrep pipeline breaks when searching for a t1 in the native subdirectory as it returns the NIfTI file instead of the MINC file. This fixes the issue and returns only the MINC file.

### This fixes...

-[x] issue #352

### CODE REVIEW

File: DTIPrep/DTI/DTI.pm [hunk 1]
```diff
@@ -162,14 +162,14 @@ RETURNS:
 sub getAnatFile {
     my ($nativedir, $t1_scan_type)  = @_;
 
-    # Fetch files in native directory that matched t1_scan_type
-    my $anat_list   = DTI::getFilesList($nativedir, $t1_scan_type);
+    # Fetch files in native directory that matched t1_scan_type and MINC file type
+    my $anat_list = DTI::getFilesList($nativedir, "\_$t1_scan_type\_.*mnc\$");
```
Comment (nicolasbrossard) [inline DTIPrep/DTI/DTI.pm]: Shouldn't the pattern passed as argument be

`"\_$t1_scan_type\_*.mnc\$"`

(watch the position of the `*`)?

Comment (cmadjar) [reply DTIPrep/DTI/DTI.pm]: this is a regex and if I understand correctly `.*` means that it matches any characters 0 or more times between the `$t1_scan_type` and `mnc` file extension.

Comment (nicolasbrossard) [reply DTIPrep/DTI/DTI.pm]: Actually, I think we want to check that the file name ends with this pattern:

1. An underscore, followed by
2. The scan type followed by
3. An underscore, followed by
4. Any non-zero length sequence of characters (except underscore, as these are used as delimiters in a MINC file name) and finally
5. The string `.mnc`

This is written as `"\_$t1_scan_type\_[^_]+\.mnc\$"`.

Makes sense?

File: DTIPrep/DTI/DTI.pm [hunk 2]
```diff
@@ -162,14 +162,14 @@ RETURNS:
...
 
     # Return undef if no anat found, first anat otherwise
     if (@$anat_list == 0) { 
         return undef; 
     } else { 
-        my $anat    = @$anat_list[0];
+        my $anat = @$anat_list[0];
```
Comment (nicolasbrossard) [inline DTIPrep/DTI/DTI.pm]: Could be written as 

```perl
return @$anat_list ? @$anat_list[0] : undef;
```

Comment (cmadjar) [reply DTIPrep/DTI/DTI.pm]: Very good point. Should have thought of that...

**Comment by nicolasbrossard** [comment]
I noticed that you dismissed my suggestion for the regex change. Is it that you don't agree with it or just that you forgot to implement the change?

**Comment by cmadjar** [comment]
> I noticed that you dismissed my suggestion for the regex change. Is it that you don't agree with it or just that you forgot to implement the change?

@nicolasbrossard so weird. I thought I implemented it... This is done now :).

**Comment by nicolasbrossard** [comment]
LGTM . 👍

**Pull request merged**